### PR TITLE
Allow a strategy to move several die rolls at once

### PR DESCRIPTION
**Pull request by jacoblever**
Also move all 'is this move possible' logic into the Game class

### CODE REVIEW

File: src/board.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def move_piece(self, piece, die_roll):
             piece_to_take.location = self.__taken_location(piece_to_take.colour)
 
         piece.location = new_location
+        return new_location
```
Comment (jacoblever) [inline src/board.py:87]: This is the `move_piece` method.

We now return the new location of the piece that was moved. This is just helpful, as it avoids callers of this class having to work it out themselves

File: src/game.py [hunk 2]
```diff
@@ -90,13 +94,29 @@ def handle_move(location, die_roll):
...
-                if self.board.is_move_possible(piece, die_roll):
-                    self.board.move_piece(piece, die_roll)
-                    break
+    def get_rolls_to_move(self, location, requested_move, available_rolls):
+        if available_rolls.__contains__(requested_move):
+            if self.board.is_move_possible(self.board.get_piece_at(location), requested_move):
+                return [requested_move]
+            return None
+        if len(available_rolls) == 1:
+            return None
```
Comment (jacoblever) [inline src/game.py:105]: The first bit of this function isn't really needed. However I'm optimising for the case where the requested_move is one of the rolls (and avoiding the need to copy the board). I'm doing this to keep experiments as fast as possible

File: app.py [hunk 3]
```diff
@@ -29,8 +29,9 @@ def move(self, board, colour, dice_roll, make_move):
                 if move == 'end_game':
                     raise Exception()
                 try:
-                    make_move(move['location'], move['die_roll'])
-                    dice_roll.remove(move['die_roll'])
+                    rolls_moved = make_move(move['location'], move['die_roll'])
+                    for roll in rolls_moved:
+                        dice_roll.remove(roll)
```
Comment (jacoblever) [inline app.py:34]: This is to centralise the logic for this change.

We now give `make_move` to total move we want to make, and it returns which dice_rolls were actually used to carry that out.

File: src/game.py [hunk 4]
```diff
@@ -52,11 +53,14 @@ def run_game(self, verbose=True):
                 print("%s rolled %s" % (colour, dice_roll))
 
             def handle_move(location, die_roll):
-                if not dice_roll.__contains__(die_roll):
-                    raise Exception("%d is not a roll that's allowed" % die_roll)
-                piece = self.board.get_piece_at(location)
-                self.board.move_piece(piece, die_roll)
-                dice_roll.remove(die_roll)
+                rolls_to_move = self.get_rolls_to_move(location, die_roll, dice_roll)
```
Comment (aknandi) [inline src/game.py:56]: Should `dice_roll` not be passed to the handle_move function?

**Pull request merged**